core/src/main/java/org/sonarsource/sonarlint/core/container/global/NodeJsHelper.java

```diff
@@ -0,0 +1,185 @@
+  private final AbstractGlobalConfiguration config;
+  private final Path pathHelperLocationOnMac;
+
+  private Path nodeJsPath;
+  private Version nodeJsVersion;
+
+  public NodeJsHelper(AbstractGlobalConfiguration config, System2 system2) {
+    this(config, system2, ProcessBuilder::new, Paths.get("/usr/libexec/path_helper"));
+  }
+
+  // For testing
+  NodeJsHelper(AbstractGlobalConfiguration config, System2 system2, Supplier<ProcessBuilder> processBuilderFactory, Path pathHelperLocationOnMac) {
+    this.config = config;
+    this.system2 = system2;
+    this.processBuilderFactory = processBuilderFactory;
+    this.pathHelperLocationOnMac = pathHelperLocationOnMac;
+  }
+
+  @Override
+  public void start() {
+    nodeJsPath = locateNode(config);
+    if (nodeJsPath != null) {
+      LOG.debug("Checking node version...");
+      String nodeVersionStr = run(Optional.empty(), nodeJsPath.toString(), "-v");
+      if (nodeVersionStr != null) {
+        Matcher matcher = NODEJS_VERSION_PATTERN.matcher(nodeVersionStr);
+        if (matcher.matches()) {
+          String version = matcher.group(1);
+          nodeJsVersion = Version.create(version);
+        }
+      }
+      if (nodeJsVersion == null) {
+        LOG.warn("Unable to query node version");
+      }
+    }
+  }
+
+  @CheckForNull
+  public Path getNodeJsPath() {
+    return nodeJsPath;
+  }
+
+  @CheckForNull
+  public Version getNodeJsVersion() {
+    return nodeJsVersion;
+  }
+
+  @CheckForNull
+  private Path locateNode(AbstractGlobalConfiguration config) {
+    Path configuredNodejsPath = config.getNodejsPath();
+    if (configuredNodejsPath != null) {
+      LOG.debug("Node.js path provided by configuration: {}", configuredNodejsPath);
+      return configuredNodejsPath;
+    }
+    LOG.debug("Looking for node in the PATH");
+
+    String result;
+    if (system2.isOsWindows()) {
+      result = run(Optional.empty(), "where", "node");
+    } else if (system2.isOsMac()) {
+      String pathEntries = null;
+      if (Files.exists(pathHelperLocationOnMac)) {
+        String pathHelperOutput = run(Optional.empty(), pathHelperLocationOnMac.toString(), "-s");
+        Pattern regex = Pattern.compile(".*PATH=\"(.*)\"; export PATH;.*");
+        Matcher matchResult = regex.matcher(pathHelperOutput);
+        if (matchResult.matches()) {
+          pathEntries = matchResult.group(1);
+        }
+      }
+      result = run(Optional.ofNullable(pathEntries), "which", "node");
+    } else {
+      result = run(Optional.empty(), "which", "node");
+    }
+    if (result != null) {
+      LOG.debug("Found node at {}", result);
+      return Paths.get(result);
+    } else {
+      LOG.debug("Unable to locate node");
+      return null;
+    }
+  }
+
+  /**
+   * Run a simple command and return stdout.
+   * @param pathEnv Override PATH
+   * @return stdout
+   */
+  @CheckForNull
+  private String run(Optional<String> pathEnv, String... command) {
```

Comment:
I would extract this method to a utility method / class, to separate the responsibility of actually running the command from the parsing of stdout.